AnkiDroid/src/main/java/com/ichi2/anki/AddonBrowser.kt

```diff
@@ -0,0 +1,173 @@
+    private lateinit var mAddonsListRecyclerView: RecyclerView
+    private val GET_ADDONS = "get_addons"
+    private var mBooleanExtra: Boolean = false
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        if (showedActivityFailedScreen(savedInstanceState)) {
+            return
+        }
+        super.onCreate(savedInstanceState)
+        Timber.d("onCreate()")
+        setContentView(R.layout.addons_browser)
+        initNavigationDrawer(findViewById(android.R.id.content))
+
+        // Add a home button to the actionbar
+        supportActionBar!!.setHomeButtonEnabled(true)
+        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
+        supportActionBar!!.title = getString(R.string.js_addons)
+        showBackIcon()
+
+        mAddonsListRecyclerView = findViewById(R.id.addons)
+        mAddonsListRecyclerView.layoutManager = LinearLayoutManager(this)
+
+        hideProgressBar()
+
+        // if activity is created from Addons Browser or Get Addons
+        mBooleanExtra = intent.getBooleanExtra(GET_ADDONS, false)
+        if (mBooleanExtra) {
+            fetchAddonsPackageJson()
+        } else {
+            listAddonsFromDir()
+        }
+    }
+
+    /**
+     * It adds a **Get Addons** menu button to Addon Browser screen
+     *
+     * @param menu
+     * @return true
+     */
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.addon_browser, menu)
+        val getMoreAddons = menu.findItem(R.id.action_get_more_addons)
+
+        getMoreAddons.setOnMenuItemClickListener {
+            // clear items from recycler view with empty list
+            mAddonsList.clear()
+            mAddonsListRecyclerView.adapter?.notifyDataSetChanged()
+
+            val intent = Intent(this, AddonBrowser::class.java)
+            intent.putExtra(GET_ADDONS, true)
+            startActivityWithAnimation(intent, ActivityTransitionAnimation.Direction.START)
+            true
+        }
+        return super.onCreateOptionsMenu(menu)
+    }
+
+    /**
+     * List addons from directory, for valid package.json the addons will be added to view
+     */
+    private fun listAddonsFromDir() {
+        val currentAnkiDroidDirectory = CollectionHelper.getCurrentAnkiDroidDirectory(this)
+
+        // AnkiDroid/addons/
+        val addonsDir = File(currentAnkiDroidDirectory, "addons")
+        if (!addonsDir.exists()) {
+            addonsDir.mkdirs()
+        }
+
+        Timber.d("List addon from directory.")
+
+        mAddonsList = ArrayList()
+        try {
+            val files = addonsDir.listFiles()
+            for (file in files!!) {
+                Timber.d("Addons: %s", file.name)
+                val mapper = ObjectMapper()
+                    .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+
+                val addonModel = mapper.readValue(File(file, "package/package.json"), AddonModel::class.java)
```

Comment:
Using jackson kotlin module, you can omit the last parameter.

```suggestion
                val addonModel: AddonModel = mapper.readValue(File(file, "package/package.json"))
```

Reply:
It showing error.

Reply:
You need to use `jackson-module-kotlin` library,

https://github.com/FasterXML/jackson-module-kotlin

Reply:
Don't forget to register it in AnkiSerilization class
```kotloin
val mapper = ObjectMapper().registerModule(KotlinModule())
```

Reply:
I understood it but now I think I should split this PR into multiple PR for faster review. Please review the other PR I will create.